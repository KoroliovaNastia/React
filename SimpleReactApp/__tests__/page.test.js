import React from 'react';
import { shallow, configure, mount, render } from 'enzyme';
import SearchPage from '../src/components/searchPage';
import DescriptionPage from '../src/components/descriptionPage';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
//import defaultImage from '../src/images/movie-default.jpg';
//import headerImage from '../src/images/header-image.jpg';


configure({ adapter: new Adapter() });

const logo = 'netflixroulete';
const searchButtonText = "Search";
const movieList = [{id: '1', image: '../src/images/movie-default.jpg', title:'Not yet...', genre:'Adventure', release_date: '2022', rating: '5.0', duration: '154', description:'lorem ipsum lorem ipsum', isShowing: true}, 
                  {id: '2', image: '../src/images/movie-default.jpg', title:'Soon...', genre:'Horror', release_date: '2020', rating: '4.1', duration: '154', description:'lorem ipsum lorem ipsum', isShowing: true}];

const filters = {
                searchFilterInfo: {type: "searchFilterInfo", title : "search by", buttonList: [{id: "b1", text: "title", checked: true}, {id: "b2", text: "genre", checked: false}]},
                navigationFilterInfo: {type: "navigationFilterInfo", title: "sort by", buttonList: [{id: "f1", text: "Release date", checked: true}, {id: "f2", text: "rating", checked: false}]}
      }

const updateFilterButtons = jest.fn();
const setIsShowing = jest.fn();

describe('SearchPage component', () => {
    it('should render correctly', () => {
        const component = shallow(<SearchPage movieList={movieList} logo={logo} filters={filters} searchButtonText={searchButtonText} updateFilterButtons={updateFilterButtons} setIsShowing={setIsShowing}/>);
        expect(shallowToJson(component)).toMatchSnapshot();
    })
})

describe('DescriptionPage component', () => {
    it('should render correctly', () => {
        const component = shallow(<DescriptionPage logo={logo} movieId={"1"} movieList={movieList} setIsShowing={setIsShowing}/>);
        expect(shallowToJson(component)).toMatchSnapshot();
    })
})