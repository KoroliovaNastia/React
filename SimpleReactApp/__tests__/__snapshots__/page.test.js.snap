// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionPage component should render correctly 1`] = `
<Fragment>
  <Header>
    <Logo
      logo="netflixroulete"
    />
    <button />
    <MovieDescription
      movie={
        Object {
          "description": "lorem ipsum lorem ipsum",
          "duration": "154",
          "genre": "Adventure",
          "id": "1",
          "image": "../src/images/movie-default.jpg",
          "isShowing": true,
          "rating": "5.0",
          "release_date": "2022",
          "title": "Not yet...",
        }
      }
    />
  </Header>
  <Box>
    <p>
      Films by 
      Adventure
       genre
    </p>
  </Box>
  <Main
    movieList={
      Array [
        undefined,
      ]
    }
  />
  <Footer>
    <Box>
      <Logo
        logo="netflixroulete"
      />
    </Box>
  </Footer>
</Fragment>
`;

exports[`SearchPage component should render correctly 1`] = `
<Fragment>
  <Header>
    <Logo
      logo="netflixroulete"
    />
    <Search
      filterInfo={
        Object {
          "buttonList": Array [
            Object {
              "checked": true,
              "id": "b1",
              "text": "title",
            },
            Object {
              "checked": false,
              "id": "b2",
              "text": "genre",
            },
          ],
          "title": "search by",
          "type": "searchFilterInfo",
        }
      }
      handleChange={[Function]}
      movieList={
        Array [
          Object {
            "description": "lorem ipsum lorem ipsum",
            "duration": "154",
            "genre": "Horror",
            "id": "2",
            "image": "../src/images/movie-default.jpg",
            "isShowing": true,
            "rating": "4.1",
            "release_date": "2020",
            "title": "Soon...",
          },
          Object {
            "description": "lorem ipsum lorem ipsum",
            "duration": "154",
            "genre": "Adventure",
            "id": "1",
            "image": "../src/images/movie-default.jpg",
            "isShowing": true,
            "rating": "5.0",
            "release_date": "2022",
            "title": "Not yet...",
          },
        ]
      }
      onSearchClick={[Function]}
      searchButtonText="Search"
      updateFilterButtons={[MockFunction]}
    />
  </Header>
  <Box>
    <p>
      2
       movie found
    </p>
    <SearchNavigation
      filterInfo={
        Object {
          "buttonList": Array [
            Object {
              "checked": true,
              "id": "f1",
              "text": "Release date",
            },
            Object {
              "checked": false,
              "id": "f2",
              "text": "rating",
            },
          ],
          "title": "sort by",
          "type": "navigationFilterInfo",
        }
      }
      updateFilterButtons={[MockFunction]}
    />
  </Box>
  <Main
    movieList={
      Array [
        Object {
          "description": "lorem ipsum lorem ipsum",
          "duration": "154",
          "genre": "Horror",
          "id": "2",
          "image": "../src/images/movie-default.jpg",
          "isShowing": true,
          "rating": "4.1",
          "release_date": "2020",
          "title": "Soon...",
        },
        Object {
          "description": "lorem ipsum lorem ipsum",
          "duration": "154",
          "genre": "Adventure",
          "id": "1",
          "image": "../src/images/movie-default.jpg",
          "isShowing": true,
          "rating": "5.0",
          "release_date": "2022",
          "title": "Not yet...",
        },
      ]
    }
  />
  <Footer>
    <Box>
      <Logo
        logo="netflixroulete"
      />
    </Box>
  </Footer>
</Fragment>
`;
